tsdb/head.go

```diff
@@ -258,7 +271,7 @@ func (h *Head) PostingsCardinalityStats(statsByLabelName string) *index.Postings
 // stripeSize sets the number of entries in the hash map, it must be a power of 2.
 // A larger stripeSize will allocate more memory up-front, but will increase performance when handling a large number of series.
 // A smaller stripeSize reduces the memory allocated, but can decrease performance with large number of series.
-func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, chunkRange int64, stripeSize int) (*Head, error) {
+func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, chunkRange int64, chkDirRoot string, pool chunkenc.Pool, stripeSize int) (*Head, error) {
```

Comment:
As an arg so that WAL need not be initialised. WAL can be nil in the tests.

And about chunk pool, yes, the pool is actually shared between block querier and compactor currently. The sad part is, nothing is being put in the pool yet, so there is no re-use it. I made a small attempt to make it re-use the chunks, but it is not trivial, so had halted my attempt.

Reply:
hm, yeah I see what you mean. I am now thinking should the chunks even be in the wal directory. Maybe it would be much better that these are in the root directory.

Reply:
Sound fine to keep it in the root directory as we always need mmapping and cannot be disabled if WAL is disabled.